File: backend/src/main/java/com/example/demo/service/PublicationService.java
package com.example.demo.service;

import com.example.demo.dto.request.PublicationRequestDTO;
import com.example.demo.dto.response.PublicationResponseDTO;

import java.util.List;

public interface PublicationService {

    PublicationResponseDTO createPublication(PublicationRequestDTO request);

    PublicationResponseDTO getPublicationById(Long id);

    List<PublicationResponseDTO> getAllPublications();

    PublicationResponseDTO updatePublication(Long id, PublicationRequestDTO request);

    void deletePublication(Long id);
    
    List<PublicationResponseDTO>
    searchPublications(String keyword);
}

File: frontend/src/services/publicationService.js
import api from "../api/api";

export const getAllPublications = async () => {
    const response = await api.get("/publications");
    return response.data;
};

export const getPublicationById = async (id) => {
    const response = await api.get(`/publications/${id}`);
    return response.data;
};

export const createPublication = async (data) => {
    const response = await api.post("/publications", data);
    return response.data;
};

export const updatePublication = async (id, data) => {
    const response = await api.put(`/publications/${id}`, data);
    return response.data;
};

export const deletePublication = async (id) => {
    const response = await api.delete(`/publications/${id}`);
    return response.data;
};
